### ISSUE TITLE
layers.set_weights() not working with tf.keras.optimizers.Ftrl

### ISSUE THREAD

**Issue by gmlu** [tensorflow/tensorflow#40086]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 10.13.2
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): ('v2.1.0-rc2-17-ge5bf8de410', '2.1.0')
- Python version: Python 2.7.14
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**
I create a model and manually initialize its weights to W0 using layers.set_weights().
Then I fit the model with tf.keras.optimizers.Ftrl and a tiny learning rate (1e-5).
The model weights became to small values around zero. It seems that model weights have been reinitialized to zeros.

**Describe the expected behavior**
Because the learning rate is very small, the model weights are expected to remain near to W0 (the manually initialized value) after training.
I tried other optimizers (SGD, Adam) from tf.keras.optimizers, the behavior is under expectation.

**Standalone code to reproduce the issue**
```python
import numpy as np
import tensorflow as tf

vocabulary = ['word1', 'word2']
categorical_column = tf.feature_column.categorical_column_with_vocabulary_list(
    'feat1', vocabulary)
embedding_column = tf.feature_column.embedding_column(
    categorical_column,
    dimension=1,
    initializer=tf.constant_initializer(0))
feature_columns = [embedding_column]

# In this model, only embeddings are trainable variables
model = tf.keras.Sequential([
    tf.keras.layers.DenseFeatures(feature_columns),
    tf.keras.layers.Dense(
        units=1,
        use_bias=False,
        trainable=False,
        kernel_initializer=tf.constant_initializer(1))
])

instances = {'feat1': np.array(['word1', 'word1', 'word2', 'word2'])}
labels = np.array([0, 0, 0, 0])

# Call the model to make variable initilized
model(instances)
print(model.trainable_variables)   # output weight [0, 0]

# Manually initialize mdoel weights to [-9, -7]
weights = [np.array([[-9], [-7]])]
model.layers[0].set_weights(weights)
print(model.trainable_variables)   # output weight [-9, -7]

# Fit the model using Ftrl optimizer with a small learning rate
optimizer = tf.keras.optimizers.Ftrl(learning_rate=1e-5)
model.compile(
    optimizer=optimizer,
    loss=tf.keras.losses.BinaryCrossentropy(from_logits=True))
model.fit(instances, labels, epochs=1, verbose=2)
print(model.trainable_variables)  # output weight [-8.501399e-07, -7.271125e-06]
```
**Other info / logs** 
```
[<tf.Variable 'sequential_12/dense_features_12/feat1_embedding/embedding_weights:0' shape=(2, 1) dtype=float32, numpy=
array([[0.],
       [0.]], dtype=float32)>]
[<tf.Variable 'sequential_12/dense_features_12/feat1_embedding/embedding_weights:0' shape=(2, 1) dtype=float32, numpy=
array([[-9.],
       [-7.]], dtype=float32)>]
Train on 4 samples
4/4 - 0s - loss: 5.1743e-04
[<tf.Variable 'sequential_12/dense_features_12/feat1_embedding/embedding_weights:0' shape=(2, 1) dtype=float32, numpy=
array([[-8.501399e-07],
       [-7.271125e-06]], dtype=float32)>]
```

**Comment by Saduf2019** [comment]
i am able to replicate this issue, please find the [gist here](https://colab.sandbox.google.com/gist/Saduf2019/064e0a3e3fd6abc3ca3f4ab766ffd834/untitled205.ipynb)

**Comment by Saduf2019** [comment]
@gmlu 
The code produces warnings on tf 2.5 which would guide you to fix the issue,
please refer to the [gist here](https://colab.research.google.com/gist/Saduf2019/d4c09f03af4c5655e88d3fc41a7c59b0/untitled597.ipynb) and upgrade your tf and let us know.
